sourcegraph/sourcegraph
# IDE extension CTAs

Resolves #27948 (except pings)

## Context

Adding a new **CTA** (little banner with a call-to-action) to boast with our cool **IDE extensions** and nudge users to try them.

This is how the new CTA should look like (also, [Figma is here](https://www.figma.com/file/VlpvIoBXrG6GFWsEFmc6rB/Browser-extension-CTA-improvements?node-id=71%3A8094)): 

![image](https://user-images.githubusercontent.com/2552265/151786174-b7fb299f-7f49-40f7-9cdf-4e2aa33e04b7.png)

The CTA should be displayed on the **search** and **file** pages for both **unauthenticated** and **signed-in** users.

Not to overwhelm our users, we're showing these at only **every 6th page load**. To avoid collision with our other CTAs (which are now also displayed at every 6th page load), we **shift** these by three page loads.

## How to test

- Search for anything, check if you see the CTA. You should see it at the 3rd, 9th, ... page loads. (You can edit the numeric values in the localStorage to avoid manual refreshes)
- Open any file and start reloading the page. You should see the CTA at the 3rd, 9th, ... page loads.
- You should see the CTA on both pages both signed in and signed out.
- The CTA should look like the one in the Figma file.
- When clicking the link, you should be taken to a relevant page where you can learn about and download the IDE extensions.
- If you dismiss the CTA (top-right X), you shouldn't see it again.
